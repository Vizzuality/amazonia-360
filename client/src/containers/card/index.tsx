"use client";

import { PropsWithChildren, MouseEvent } from "react";

import Image from "next/image";

import { UseQueryResult } from "@tanstack/react-query";
import { LuInfo, LuSettings2 } from "react-icons/lu";

import { cn } from "@/lib/utils";

import { DatasetIds } from "@/constants/datasets";

import Info from "@/containers/info";

import { Dialog, DialogContent, DialogTrigger } from "@/components/ui/dialog";
import { Skeleton } from "@/components/ui/skeleton";
import { Tooltip, TooltipArrow, TooltipContent, TooltipTrigger } from "@/components/ui/tooltip";

interface CardProps {
  padding?: boolean;
  className?: string;
}

export function CardHeader({
  className,
  children,
}: PropsWithChildren<{
  className?: string;
}>) {
  return <header className={cn("flex items-start justify-between", className)}>{children}</header>;
}

export function CardTitle({ children }: PropsWithChildren) {
  return <h2 className="text-base font-semibold text-blue-600">{children}</h2>;
}

export function CardContent({ className, children }: PropsWithChildren<{ className?: string }>) {
  return <div className={cn("mt-2 flex grow flex-col", className)}>{children}</div>;
}

export function CardSettings({
  id,
  onClick,
}: PropsWithChildren<{ id?: string; onClick?: (e: MouseEvent<HTMLElement>) => void }>) {
  return (
    <Tooltip delayDuration={100}>
      <TooltipTrigger asChild>
        <button
          id={id}
          type="button"
          className="text-base font-semibold text-blue-600"
          onClick={onClick}
        >
          <LuSettings2 className="text-blue-600" />
        </button>
      </TooltipTrigger>

      <TooltipContent sideOffset={0}>
        Edit indicator
        <TooltipArrow />
      </TooltipContent>
    </Tooltip>
  );
}

export function CardControls({ children }: PropsWithChildren) {
  return <div className="flex space-x-2">{children}</div>;
}

export function CardInfo({ ids, className }: { ids: DatasetIds[]; className?: string }) {
  return (
    <Tooltip delayDuration={100}>
      <Dialog>
        <TooltipTrigger asChild>
          <DialogTrigger className={cn("flex h-6 w-6 items-center justify-center", className)}>
            <LuInfo className="text-blue-600" />
          </DialogTrigger>
        </TooltipTrigger>
        <DialogContent className="p-0">
          <Info ids={ids} />
        </DialogContent>
        <TooltipContent sideOffset={0}>
          About the data
          <TooltipArrow />
        </TooltipContent>
      </Dialog>
    </Tooltip>
  );
}

export function CardLoader({
  query,
  children,
  ...rest
}: {
  query: UseQueryResult<unknown, unknown>[];
  children: React.ReactNode;
} & React.HTMLAttributes<HTMLDivElement>) {
  if (query.some((q) => q.isFetching)) {
    return <Skeleton data-testid="card-loader" {...rest} />;
  }

  return <>{children}</>;
}

export function CardNoData({
  query,
  children,
}: {
  query: UseQueryResult<unknown, unknown>[];
  children: React.ReactNode;
}) {
  if (
    query.every((q) => q.isFetched) &&
    !query.some((q) => q.data && Array.isArray(q.data) && !!q.data.length)
  ) {
    return (
      <div className="flex grow flex-col items-center justify-center space-y-6 py-12">
        <Image src={"/images/no-data.svg"} alt="No data" width={149} height={96} />
        <p className="text-center text-sm font-medium text-blue-900">
          No results for this location at the moment. <br /> Feel free to adjust your search
          criteria or check back later!
        </p>
      </div>
    );
  }

  return <>{children}</>;
}
export function CardWidgetNumber({
  value,
  subvalue,
  unit,
}: {
  value: string | number | null;
  subvalue?: string | number | null;
  unit?: string;
}) {
  return (
    <div>
      <div className="flex items-end space-x-2">
        <span className="text-4xl font-bold text-blue-600">{value}</span>

        {!!unit && (
          <span className="relative bottom-1.5 text-xs font-medium text-gray-500">{unit}</span>
        )}
      </div>

      {subvalue && <p className="text-xs text-foreground">{subvalue}</p>}
    </div>
  );
}

export function Card({ className, children }: PropsWithChildren<CardProps>) {
  return (
    <div
      className={cn(
        "flex h-full grow break-inside-avoid flex-col overflow-hidden rounded-2xl border border-blue-100 bg-white p-6",
        className,
      )}
    >
      {children}
    </div>
  );
}
