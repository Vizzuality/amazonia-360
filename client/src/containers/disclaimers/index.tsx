import { PropsWithChildren } from "react";

import { cn } from "@/lib/utils";

export default function Disclaimer({
  children,
  className,
}: PropsWithChildren<{
  className?: string;
}>) {
  return (
    <div className={cn("bg-amber-100 py-4 print:hidden", className)}>
      {children}
    </div>
  );
}
