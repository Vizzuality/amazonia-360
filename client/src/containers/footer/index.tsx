import Image from "next/image";

import { Button } from "@/components/ui/button";

export default function Help() {
  return (
    <section className="bg-blue-700 text-white print:hidden">
      <div className="container flex md:flex-row flex-col justify-between w-full py-4 items-center">
        <div className="pb-16 md:pb-0 flex md:space-x-10 flex-col space-y-10 md:space-y-0 items-center md:flex-row">
          <Image
            src={"/images/home/idb_logo.png"}
            alt="IDB Logo"
            width={65}
            height={24}
            className="text-white"
          />
          <Image
            src={"/images/home/amazonia_forever_logo.png"}
            alt="KF Logo"
            width={65}
            height={48}
            className="text-white"
          />
        </div>
        <div className="text-sm font-light flex md:flex-row flex-col">
          <Button variant="link" className="text-white leading-1">
            <a
              href="https://www.iadb.org/en/home/terms-and-conditions"
              target="_blank"
            >
              Terms & conditions
            </a>
          </Button>
          <Button variant="link" className="text-white leading-1">
            <a
              href="https://www.iadb.org/en/home/privacy-notice"
              target="_blank"
            >
              Privacy Policy
            </a>
          </Button>
        </div>
      </div>
    </section>
  );
}
