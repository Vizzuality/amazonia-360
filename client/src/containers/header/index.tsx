"use client";

import Image from "next/image";
import Link from "next/link";
import { usePathname } from "next/navigation";

import { cn } from "@/lib/utils";

import { Badge } from "@/components/ui/badge";

export default function Header() {
  const pathname = usePathname();

  return (
    <header className="flex h-20 flex-col justify-center bg-white backdrop-blur">
      <div className="container mx-auto flex items-center justify-between">
        <Link href="/">
          <h1 className="flex items-center space-x-4">
            <Image src="/IDB-logo.svg" alt="IDB" width={65} height={24} />
            <div className="space-x-2">
              <span className="text-sm font-medium">Amazonia360+</span>
              <Badge variant="secondary">Prototype</Badge>
            </div>
          </h1>
        </Link>

        <nav className="flex space-x-8 print:hidden">
          <Link
            className={cn({
              "text-sm hover:text-cyan-500": true,
              "text-cyan-500": pathname === "/",
            })}
            href="/"
          >
            Home
          </Link>

          <Link
            className={cn({
              "text-sm hover:text-cyan-500": true,
              "text-cyan-500": pathname.includes("/report"),
            })}
            href="/report"
          >
            Report Tool
          </Link>
          <Link
            className={cn({
              "text-sm hover:text-cyan-500": true,
              "text-cyan-500": pathname.includes("/hub"),
            })}
            href="/hub"
          >
            Hub
          </Link>
          <Link
            className={cn({
              "text-sm hover:text-cyan-500": true,
              "text-cyan-500": pathname.includes("/grid"),
            })}
            href="/grid"
          >
            Grid
          </Link>

          {/* <Link className="text-blue-600 hover:text-blue-700" href="#">
            Amazonia Forever
          </Link> */}
        </nav>
      </div>
    </header>
  );
}
