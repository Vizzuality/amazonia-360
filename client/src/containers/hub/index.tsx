import Image from "next/image";
import Link from "next/link";

import { LuArrowRight } from "react-icons/lu";

import { Button } from "@/components/ui/button";

export default function Hub() {
  return (
    <main className="flex flex-col-reverse md:flex-row w-full bg-blue-50 md:h-[calc(100svh_-_theme(space.40)_+_1px)] relative">
      <div className="container flex h-full md:pr-60 py-12 md:py-0">
        <div className="w-full md:w-1/2 h-full flex flex-col justify-center p-10">
          <h2 className="text-blue-400 text-3xl lg:text-4xl pb-6 font-semibold">
            We are currently <br />
            building this feature!
          </h2>
          <div className="text-blue-900 text-base font-normal pb-12 flex flex-col space-y-2">
            <p>
              We&apos;re currently developing the Hub - your one-stop
              destination for accessing and sharing all essential information to{" "}
              <b>guide decision-making</b> in the Amazonia region. Here,
              <b>collaboration, coordination,</b> and <b>knowledge sharing</b>{" "}
              empower informed choices and drive <b>positive impact</b>.
            </p>
            <p>
              While we are working on this, we encourage you to explore our{" "}
              <b>Reporting tool</b>, offering valuable insights into the
              Amazonia.
            </p>
          </div>
          <Link href="/report">
            <Button size="lg" className="mt-4 space-x-2.5 flex">
              <p>Explore reporting tool</p>
              <LuArrowRight size={20} strokeWidth={1} />
            </Button>
          </Link>
        </div>
      </div>
      <div className="md:absolute md:right-0 md:top-0 md:w-1/2 md:h-full w-full">
        <Image
          src={"/images/hub/coming-soon.jpg"}
          alt="Coming soon..."
          width={2000}
          height={2000}
          priority
          className="object-cover h-full w-full"
        />
      </div>
    </main>
  );
}
