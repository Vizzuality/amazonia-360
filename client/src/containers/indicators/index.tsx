"use client";

import { useState } from "react";

import { useQueryClient } from "@tanstack/react-query";

import {
  getQueryFeatureIdKey,
  useIndicators,
  useIndicatorsId,
  useQueryFeatureId,
} from "@/lib/indicators";

import { Indicator, ResourceFeature, VisualizationType } from "@/app/api/indicators/route";

import { ChartIndicators } from "@/containers/indicators/chart";
import { NumericIndicators } from "@/containers/indicators/numeric";
import { TableIndicators } from "@/containers/indicators/table";

import { Button } from "@/components/ui/button";

export const ResourceQueryFeature = ({
  type,
  resource,
}: Indicator & {
  type: VisualizationType;
  resource: ResourceFeature;
}) => {
  const queryClient = useQueryClient();
  const [enabled, setEnabled] = useState(false);
  const { data } = useQueryFeatureId({ resource, type }, { enabled });

  console.info(data);

  return (
    <div className="space-y-5">
      <div>
        <pre>{JSON.stringify(resource[`query_${type}`], null, 2)}</pre>
        <Button
          size="sm"
          onClick={() => {
            const key = getQueryFeatureIdKey({ resource, type });
            queryClient.invalidateQueries({
              queryKey: key,
            });
            setEnabled(true);
          }}
        >
          Run Query
        </Button>
      </div>

      <div className="not-prose">
        {type === "table" && enabled && <TableIndicators resource={resource} />}
        {type === "chart" && enabled && <ChartIndicators resource={resource} />}
        {type === "numeric" && enabled && <NumericIndicators resource={resource} />}
      </div>
    </div>
  );
};

export const IndicatorItem = ({ id }: { id: Indicator["id"] }) => {
  const indicator = useIndicatorsId(id);

  return (
    <div className="space-y-5">
      <h2 className="text-3xl">{indicator?.name}</h2>
      <div className="prose prose-sm max-w-none">
        <table className="w-full">
          <tbody>
            <tr>
              <td className="w-60">Topic</td>
              <td>
                {indicator?.topic_id} - {indicator?.topic_name}
              </td>
            </tr>
            <tr>
              <td className="w-60">Resource</td>
              <td>{indicator?.resource.name}</td>
            </tr>
            <tr>
              <td className="w-60">Resource Type</td>
              <td>{indicator?.resource.type}</td>
            </tr>
            <tr>
              <td className="w-60">Resource URL</td>
              <td>
                <a href={indicator?.resource.url} target="_blank" rel="noreferrer">
                  {indicator?.resource.url}
                </a>
              </td>
            </tr>
            <tr>
              <td className="w-60">Visualization Types</td>
              <td>{indicator?.visualization_types.join(", ")}</td>
            </tr>
            {indicator?.visualization_types.map((type) => {
              if (indicator?.resource.type === "feature") {
                const r = indicator.resource;

                return (
                  <tr key={type}>
                    <td>Query {type[0].toUpperCase() + type.slice(1)}</td>
                    <td>
                      <ResourceQueryFeature {...indicator} type={type} resource={r} />
                    </td>
                  </tr>
                );
              }

              return null;
            })}
            <tr>
              <td className="w-60">H3 Grid Column Name</td>
              <td>{indicator?.h3_grid_column_name || "-"}</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  );
};

export const Indicators = () => {
  const { data } = useIndicators({
    refetchOnWindowFocus: "always",
    refetchOnMount: "always",
    refetchOnReconnect: "always",
  });

  return (
    <section className="py-10">
      <div className="container">
        <ul className="flex flex-col gap-20">
          {data?.map((indicator) => (
            <li key={indicator.id}>
              <IndicatorItem id={indicator.id} />
            </li>
          ))}
        </ul>
      </div>
    </section>
  );
};
