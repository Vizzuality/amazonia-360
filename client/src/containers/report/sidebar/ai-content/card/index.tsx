"use client";

import { cn } from "@/lib/utils";

import { Label } from "@/components/ui/label";
import { RadioGroupItem } from "@/components/ui/radio-group";

export default function AiSidebarContentCard({
  option,
  active,
}: {
  option: { value: string; label: string; description: string };
  active: boolean;
}) {
  return (
    <Label
      htmlFor={option.value}
      className={cn({
        "flex cursor-pointer flex-col rounded-sm border p-4 text-sm transition-all duration-200 hover:border-[1.5px] hover:border-foreground":
          true,
        "border-foreground": active,
      })}
    >
      <div key={option.value} className="flex items-start space-x-4 text-foreground">
        <RadioGroupItem id={option.value} value={option.value} />
        <div className="flex flex-col">
          <span className="font-bold">{option.label}</span>
          <p className="font-medium text-muted-foreground">{option.description}</p>
        </div>
      </div>
    </Label>
  );
}
