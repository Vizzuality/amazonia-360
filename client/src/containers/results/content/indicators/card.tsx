"use client";

import { createElement, MouseEvent } from "react";

import { useGetIndicatorsId } from "@/lib/indicators";
import { cn } from "@/lib/utils";

import { Indicator, VisualizationTypes } from "@/app/local-api/indicators/route";
import {
  ResourceFeature,
  ResourceImageryTile,
  ResourceWebTile,
} from "@/app/local-api/indicators/route";

import {
  Card,
  CardContent,
  CardHeader,
  CardControls,
  CardSettings,
  CardInfo,
  CardTitle,
} from "@/containers/card";
import { ChartIndicators } from "@/containers/indicators/chart";
import { ChartImageryIndicators } from "@/containers/indicators/chart/imagery";
import { ChartImageryTileIndicators } from "@/containers/indicators/chart/imagery-tile";
import { MapIndicators } from "@/containers/indicators/map";
import { NumericIndicators } from "@/containers/indicators/numeric";
import { NumericImageryIndicators } from "@/containers/indicators/numeric/imagery";
<<<<<<< HEAD
=======
import { NumericImageryTileIndicators } from "@/containers/indicators/numeric/imagery-tile";
import { TotalArea } from "@/containers/indicators/numeric/total-area";
>>>>>>> a1d4ced (Sorting and isolated components)
import { TableIndicators } from "@/containers/indicators/table";

// custom indicators
import { TotalArea } from "@/containers/indicators/custom/total-area";
import { Municipalities } from "@/containers/indicators/custom/municipalities";

const COMPONENT_INDICATORS = {
  "total-area": TotalArea,
  AMZ_LOCADM2: Municipalities,
} as const;

type COMPONENT_INDICATORS_KEYS = keyof typeof COMPONENT_INDICATORS;

export default function ReportResultsIndicator({
  id,
  type,
  editable,
  onEdit,
}: {
  id: Indicator["id"];
  type: VisualizationTypes;
  editable: boolean;
  onEdit?: (e: MouseEvent<HTMLElement>) => void;
}) {
  const indicator = useGetIndicatorsId(id);

  if (!indicator) return null;

  return (
    <div className="flex h-full flex-col">
      <Card className={cn(type === "map" && "p-0")}>
        <CardHeader className={cn(type === "map" && "px-6 pt-6")}>
          <CardTitle>{indicator?.name_en}</CardTitle>
          <CardControls>
            <CardInfo ids={[indicator.id]} />

            {editable && <CardSettings id={indicator?.id} onClick={onEdit} />}
          </CardControls>
        </CardHeader>
        <CardContent>
          {type === "map" && indicator.resource.type !== "h3" && (
            <MapIndicators
              {...(indicator as Omit<Indicator, "resource"> & {
                resource: ResourceFeature | ResourceWebTile | ResourceImageryTile;
              })}
            />
          )}

          {/*
            Charts
          */}
          {type === "chart" && indicator.resource.type === "feature" && (
            <ChartIndicators {...indicator} resource={indicator.resource} />
          )}
          {type === "chart" && indicator.resource.type === "imagery" && (
            <ChartImageryIndicators {...indicator} resource={indicator.resource} />
          )}
<<<<<<< HEAD
          {indicator.resource.type === "component" &&
=======
          {type === "chart" && indicator.resource.type === "imagery-tile" && (
            <ChartImageryTileIndicators {...indicator} resource={indicator.resource} />
          )}

          {/*
            Numerics
          */}
          {type === "numeric" &&
            indicator.resource.type === "component" &&
>>>>>>> a1d4ced (Sorting and isolated components)
            !!COMPONENT_INDICATORS[`${indicator.resource.name}` as COMPONENT_INDICATORS_KEYS] &&
            createElement(
              COMPONENT_INDICATORS[`${indicator.resource.name}` as COMPONENT_INDICATORS_KEYS],
              { indicator },
            )}

          {type === "numeric" && indicator.resource.type === "feature" && (
            <NumericIndicators {...indicator} resource={indicator.resource} />
          )}
          {type === "numeric" && indicator.resource.type === "imagery" && (
            <NumericImageryIndicators {...indicator} resource={indicator.resource} />
          )}
          {type === "numeric" && indicator.resource.type === "imagery-tile" && (
            <NumericImageryTileIndicators {...indicator} resource={indicator.resource} />
          )}

          {/*
            Table
          */}
          {type === "table" && indicator.resource.type === "feature" && (
            <TableIndicators {...indicator} resource={indicator.resource} />
          )}
        </CardContent>
      </Card>
    </div>
  );
}
