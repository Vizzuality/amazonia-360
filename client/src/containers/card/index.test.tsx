import { UseQueryResult } from "@tanstack/react-query";
import { render, screen } from "@testing-library/react";

import { CardLoader, CardNoData } from "@/containers/card";

describe("CardLoader", () => {
  it("renders CardLoader component without crashing", () => {
    const { container } = render(
      <CardLoader query={[{ isFetching: true } as UseQueryResult<unknown, unknown>]}>
        <div role="presentation">Test</div>
      </CardLoader>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders CardLoader Skeleton component when query is fetching", () => {
    const { container } = render(
      <CardLoader query={[{ isFetching: true } as UseQueryResult<unknown, unknown>]}>
        <div role="presentation">Test</div>
      </CardLoader>,
    );
    expect(container).toMatchSnapshot();
    expect(screen.queryByRole("presentation")).not.toBeInTheDocument();
  });

  it("renders CardLoader Children component when query is not fetching", () => {
    const { container } = render(
      <CardLoader query={[{ isFetching: false } as UseQueryResult<unknown, unknown>]}>
        <div role="presentation">Test</div>
      </CardLoader>,
    );
    expect(container).toMatchSnapshot();
    expect(screen.queryByRole("presentation")).toBeInTheDocument();
  });
});

describe("CardNoData", () => {
  it("renders CardNoData component without crashing", () => {
    const { container } = render(
      <CardNoData query={[{ isFetched: true } as UseQueryResult<unknown, unknown>]}>
        <div role="presentation">Test</div>
      </CardNoData>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders CardNoData component when query is fetched and no data", () => {
    const { container } = render(
      <CardNoData query={[{ isFetched: true, data: [] } as UseQueryResult<unknown, unknown>]}>
        <div role="presentation">Test</div>
      </CardNoData>,
    );
    expect(container).toMatchSnapshot();
    expect(screen.queryByRole("presentation")).not.toBeInTheDocument();
  });

  it("renders CardNoData Children component when query is fetched and there is data", () => {
    const { container } = render(
      <CardNoData
        query={[{ isFetched: true, data: [{ value: 1 }] } as UseQueryResult<unknown, unknown>]}
      >
        <div role="presentation">Test</div>
      </CardNoData>,
    );
    expect(container).toMatchSnapshot();
    expect(screen.queryByRole("presentation")).toBeInTheDocument();
  });
});
