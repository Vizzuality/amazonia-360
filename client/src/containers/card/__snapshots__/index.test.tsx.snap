// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLoader renders CardLoader Children component when query is not fetching 1`] = `
<div>
  <div
    role="presentation"
  >
    Test
  </div>
</div>
`;

exports[`CardLoader renders CardLoader Skeleton component when query is fetching 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-primary/10"
    data-testid="card-loader"
  />
</div>
`;

exports[`CardLoader renders CardLoader component without crashing 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-primary/10"
    data-testid="card-loader"
  />
</div>
`;

exports[`CardNoData renders CardNoData Children component when query is fetched and there is data 1`] = `
<div>
  <div
    role="presentation"
  >
    Test
  </div>
</div>
`;

exports[`CardNoData renders CardNoData component when query is fetched and no data 1`] = `
<div>
  <div
    class="flex grow flex-col items-center justify-center space-y-6 py-12"
  >
    <img
      alt="No data"
      data-nimg="1"
      decoding="async"
      height="96"
      loading="lazy"
      src="/images/no-data.svg"
      style="color: transparent;"
      width="149"
    />
    <p
      class="text-center text-sm font-medium text-blue-900"
    >
      No results for this location at the moment. 
      <br />
       Feel free to adjust your search criteria or check back later!
    </p>
  </div>
</div>
`;

exports[`CardNoData renders CardNoData component without crashing 1`] = `
<div>
  <div
    class="flex grow flex-col items-center justify-center space-y-6 py-12"
  >
    <img
      alt="No data"
      data-nimg="1"
      decoding="async"
      height="96"
      loading="lazy"
      src="/images/no-data.svg"
      style="color: transparent;"
      width="149"
    />
    <p
      class="text-center text-sm font-medium text-blue-900"
    >
      No results for this location at the moment. 
      <br />
       Feel free to adjust your search criteria or check back later!
    </p>
  </div>
</div>
`;
